ethereum_history_api/circuits/lib/src/verifiers/receipt/rlp.nr

```diff
@@ -0,0 +1,68 @@
+use crate::{
+    verifiers::receipt::{MAX_RECEIPT_RLP_LENGTH, is_pre_byzantium},
+    receipt::{BLOOM_FILTER_LEN, TxReceiptPartial}
+};
+use dep::proof::{const::HASH_LENGTH, rlp::{RLP_List, decode1, STRING}, utils::byte_value};
+use crate::misc::arrays::sub_array_equals_up_to_length;
+use dep::u2b::u32_to_u8;
+
+global RECEIPT_FIELDS_COUNT = 4;
+global STATE_ROOT_INDEX = 0;
+global STATUS_INDEX = 0;
+global CUMULATIVE_GAS_USED_INDEX = 1;
+global LOGS_BLOOM_INDEX = 2;
+
+pub(crate) fn assert_receipt_rlp_equals<LOG_NUM, MAX_LOG_DATA_LEN>(
+    block_number: Field,
```

Comment:
Question:
Should we pass `block_number` or `is_pre_byzantium` or fork designation or header.
Really not sure, but it feels like a wrong level of abstraction.

Reply:
Fixed